### PR TITLE
Oceanwater 479 publish mechanical power arm

### PR DESCRIPTION
**Pull request by anjan1984**
This node publishes transient mechanical power on two topics:  /mechanical_power/raw and /mechanical_power/average

### CODE REVIEW

File: ow_lander/launch/planning.launch [hunk 1]
```diff
@@ -49,6 +49,9 @@
     <!-- == launch the trajectory publishing node ========= -->
     <node pkg="ow_lander" name="trajectory_publisher" type="trajectory_publisher.py" output="screen">
     </node>
+    <!-- == launch the mechanical power publishing node ========= -->
+    <node pkg="ow_lander" name="mechanical_power_arm" type="mechanical_power_arm.py" output="screen">
```
Comment (mogumbo) [inline ow_lander/launch/planning.launch:53]: This is unrelated to planning. Shouldn't it be in common.launch instead of planning.launch?

File: ow_lander/scripts/mechanical_power_arm.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
```
Comment (mogumbo) [inline ow_lander/scripts/mechanical_power_arm.py:1]: Change python to python2. We need to be specific for users who have python3 configured as the default on their computers.

File: ow_lander/scripts/mechanical_power_arm.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+# Python libs
+import sys, time
+import numpy as np
+
+#history = []
+
+VERBOSE=True
+
+class MechanicalPower:
+    def __init__(self):
```
Comment (mogumbo) [inline ow_lander/scripts/mechanical_power_arm.py:16]: 2-space indentation please.

File: ow_lander/scripts/mechanical_power_arm.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            self.history = self.history[-10:]
+            average = sum(self.history) / float(len(self.history))
+            #rospy.loginfo('Average of most recent {} samples: {}'.format(len(self.history), average))
+        else:
+            average = msg
+        return average    
+        
+    def callback(self, ros_data):
+        self._value = len(ros_data.name)
+        l = len(ros_data.name)
```
Comment (mogumbo) [inline ow_lander/scripts/mechanical_power_arm.py:42]: "l" isn't much of a variable name. How about "num_joints" or something descriptive?

**Review by mogumbo — CHANGES_REQUESTED** [review]

File: ow_lander/scripts/mechanical_power_arm.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+import rospy
+from sensor_msgs.msg import JointState
+from std_msgs.msg import Float64
+
+# Python libs
+import sys, time
+import numpy as np
+
+#history = []
```
Comment (Samahu) [inline ow_lander/scripts/mechanical_power_arm.py:11]: This seems left out
```suggestion

```

File: ow_lander/scripts/mechanical_power_arm.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+class MechanicalPower:
+    def __init__(self):
+        '''Initialize ros publisher, ros subscriber'''
+        # topic where we publish
+        self.power_pub = rospy.Publisher('/mechanical_power/raw', Float64, queue_size = 10)
+        self.power_pub_a = rospy.Publisher('/mechanical_power/average', Float64, queue_size = 10)
+        
+        self.subscriber = rospy.Subscriber('/joint_states', JointState, self.callback, queue_size = 1)
+        self.history = []
```
Comment (Samahu) [inline ow_lander/scripts/mechanical_power_arm.py:23]: I recommend using a _deque_ instead of a _list_, the **history** data member here grows indefinitely. It may not pose a real problem since the data size but things add up.

**Review by Samahu — CHANGES_REQUESTED** [review]
Even though I am requesting changes but feel free to proceed with the merge and address these requests later.

File: ow_lander/launch/planning.launch (hunk 1 above)
Comment (anjan1984) [reply ow_lander/launch/planning.launch]: Done!

File: ow_lander/scripts/mechanical_power_arm.py (hunk 2 above)
Comment (anjan1984) [reply ow_lander/scripts/mechanical_power_arm.py]: Done!

File: ow_lander/scripts/mechanical_power_arm.py (hunk 3 above)
Comment (anjan1984) [reply ow_lander/scripts/mechanical_power_arm.py]: Done!

File: ow_lander/scripts/mechanical_power_arm.py (hunk 4 above)
Comment (anjan1984) [reply ow_lander/scripts/mechanical_power_arm.py]: Done!

File: ow_lander/scripts/mechanical_power_arm.py (hunk 5 above)
Comment (anjan1984) [reply ow_lander/scripts/mechanical_power_arm.py]: Done!

File: ow_lander/scripts/mechanical_power_arm.py (hunk 6 above)
Comment (anjan1984) [reply ow_lander/scripts/mechanical_power_arm.py]: Thanks for the suggestion. I will do this later. I will have to get used to deque.

**Review by mogumbo — APPROVED** [review]

**Review by kmdalal — APPROVED** [review]

**Comment by Samahu** [comment]
This should have been merged to **melodic-devel**

**Comment by anjan1984** [comment]
Done merged to melodic-devel
